### PR TITLE
Add PyFMI

### PR DESCRIPTION
**Pull request by jschueller**
with assimulo and fmilib dependencies

### CODE REVIEW

**Comment by jschueller** [comment]
could someone help me figure out this error for my package fmilib for msvc:
```
conda.resolve.Unsatisfiable: The following specifications were found to be in conflict:
  - fmilib 2.0.2 vc9_0 -> feature:vc9
```

**Comment by jschueller** [comment]
@jakirkham could you help me with this error for the fmilib package ?
```
conda.resolve.Unsatisfiable: The following specifications were found to be in conflict:
  - fmilib 2.0.2 vc9_0 -> feature:vc9
```

**Comment by jschueller** [comment]
could someone help me with this error for the fmilib package ?

conda.resolve.Unsatisfiable: The following specifications were found to be in conflict:
  - fmilib 2.0.2 vc9_0 -> feature:vc9

File: recipes/fmilib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+
+requirements:
+  build:
+    - toolchain
+    - cmake
+    - python  # [win]
+
+test:
```
Comment (jakirkham) [inline recipes/fmilib/meta.yaml]: Please add the following. This should fix the build failure on AppVeyor.

```yaml

test:
  requires:
    - python {{ environ['PY_VER'] + '*' }}  # [win]
```

xref: https://github.com/conda/conda-build/issues/835

**Comment by jakirkham** [comment]
Could you also please rebase this also @jschueller ? I think some old recipes are showing up in this PR's review.

File: recipes/fmilib/meta.yaml (hunk 1 above)
Comment (jschueller) [inline recipes/fmilib/meta.yaml]: thanks!

**Comment by jschueller** [comment]
@jakirkham with vc14, pyqt 4.11.4-py35_6 from defaults overrides pyqt from conda-forge but does not properly pin qt 4 so qt 5.6 gets pulled and the matplotlib import fails

**Comment by jakirkham** [comment]
Could you please raise that at the [`matplotlib` feedstock]( https://github.com/conda-forge/matplotlib-feedstock )?

**Comment by jschueller** [comment]
@jakirkham no pyqt is properly pinned in matplotlib-feedstock, as well as qt is properly pinned in pyqt feedstock, it's pyqt from defaults channel which is faulty

**Comment by jakirkham** [comment]
Ok, then raise an issue with [Anaconda]( https://github.com/ContinuumIO/anaconda-issues ). Still it wouldn't be unreasonable for `matplotlib` to pin both `qt` and `pyqt` near term as this will affect lots of packages.

**Comment by jschueller** [comment]
https://github.com/ContinuumIO/anaconda-issues/issues/1164

**Comment by jakirkham** [comment]
Thanks.

**Comment by jschueller** [comment]
I think this is ready, someone to review ?

File: recipes/assimulo/bld.bat [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+:: --force-32bit=False --is_static=True
```
Comment (jakirkham) [inline recipes/assimulo/bld.bat]: What is this comment for? Is this left over code or do we need it for something?

File: recipes/assimulo/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+if test `uname` = "Darwin"
+then
+  SO_EXT='.dylib'
+else
+  SO_EXT='.so'
+fi
+
+PY_LIB=`find $PREFIX/lib -name libpython${PY_VER}*${SO_EXT}`
+
+#  --blas-home=${PREFIX}/lib --blas-name=openblas --lapack-home=${PREFIX}/lib --sundials-home=${PREFIX}/lib
```
Comment (jakirkham) [inline recipes/assimulo/build.sh]: Do we need to be setting the BLAS and such somewhere?

File: recipes/assimulo/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  number: 0
+  features:
+  - vc9  # [win and py27]
+  - vc10  # [win and py34]
+  - vc14  # [win and py35]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/assimulo/meta.yaml]: Please add `toolchain`.

File: recipes/assimulo/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    - python
+    - numpy
+    - cython
+    - sundials
+    - gcc  # [unix]
+  run:
+    - python
+    - numpy
```
Comment (jakirkham) [inline recipes/assimulo/meta.yaml]: Same as above. May need to append `x.x`.

File: recipes/assimulo/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+test:
+  imports:
+    - assimulo
+
+  requires:
+      - nose
+about:
+  home: http://www.jmodelica.org/assimulo
+  license: LGPL
```
Comment (jakirkham) [inline recipes/assimulo/meta.yaml]: License version? Also please add the license file to the package. Finally would suggest adding `license_family: LGPL` though it is optional.

File: recipes/fmilib/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    - python {{PY_VER}}*  # [win]
+  commands:
+    - test -f $PREFIX/include/fmilib.h                                # [unix]
+    - test -f $PREFIX/lib/libfmilib.a                                 # [unix]
+    - if not exist %PREFIX%\\Library\\lib\\fmilib.lib exit 1          # [win]
+    - if not exist %PREFIX%\\Library\\include\\fmilib.h exit 1        # [win]
+
+about:
+  home: http://www.jmodelica.org/FMILibrary
+  license: BSD
```
Comment (jakirkham) [inline recipes/fmilib/meta.yaml]: License version? Also can we package the license? Finally would suggest adding `license_family: BSD`.

File: recipes/fmilib/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
+    - toolchain
+    - cmake
+    - python  # [win]
+
+test:
+  requires:
+    - python {{PY_VER}}*  # [win]
```
Comment (jakirkham) [inline recipes/fmilib/meta.yaml]: The linter sometimes barfs with this syntax. So we may need the `environ` form.

**Comment by jakirkham** [comment]
I know this is only a few recipes, but would you mind breaking it up a bit? This will make it easier to review and accept. Maybe start with the lower level packages in this PR and work are way up?

File: recipes/assimulo/build.sh (hunk 3 above)
Comment (jschueller) [inline recipes/assimulo/build.sh]: no it's detected

File: recipes/assimulo/bld.bat (hunk 2 above)
Comment (jschueller) [inline recipes/assimulo/bld.bat]: ok, removed

File: recipes/assimulo/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  number: 0
+  features:
+  - vc9  # [win and py27]
+  - vc10  # [win and py34]
+  - vc14  # [win and py35]
+
+requirements:
+  build:
+    - python
+    - numpy
```
Comment (jschueller) [inline recipes/assimulo/meta.yaml]: yes, ok

File: recipes/assimulo/meta.yaml (hunk 6 above)
Comment (jschueller) [inline recipes/assimulo/meta.yaml]: seems license is tricky:
http://pkgs.fedoraproject.org/cgit/rpms/python-assimulo.git/tree/python-assimulo.spec

File: recipes/fmilib/meta.yaml (hunk 8 above)
Comment (jschueller) [inline recipes/fmilib/meta.yaml]: ok

**Comment by jschueller** [comment]
@jakirkham ok see #1790, #1791

**Comment by jschueller** [comment]
@jakirkham this is ready now

**Comment by scopatz** [comment]
Sweet Thanks!

**Pull request merged**